code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -38,6 +38,51 @@
 		if((m != R.module_state_1) && (m != R.module_state_2) && (m != R.module_state_3))
 			. += m
 
+/obj/item/weapon/robot_module/proc/add_module(var/obj/item/I)
+	if(istype(I, /obj/item/stack))
+		var/obj/item/stack/S = I
+
+		if(is_type_in_list(S, list(/obj/item/stack/sheet/metal, /obj/item/stack/rods, /obj/item/stack/tile/plasteel)))
+			var/datum/robot_energy_storage/metal/metstore = locate(/datum/robot_energy_storage/metal) in storages
+			if(!metstore)
+				metstore = new /datum/robot_energy_storage/metal(src)
+
+			if(S.materials[MAT_METAL])
+				S.cost = S.materials[MAT_METAL] / 4

```

Comment:
So in even more fancy, "if it leaves a recurring number, just divide"
and yes, since there's no fancy way (to my knowledge) of cheaply converting a division to it's multiplication form you have to divide for variables.